### ISSUE TITLE
Question about epsilon

### ISSUE THREAD

**Issue by JiamingSuen** [wuhuikai/DeepGuidedFilter#6]
Hi @wuhuikai , thanks for this amazing work.
I'm working on a project with a need for disparity estimation using CNN, and I'm trying to adapt your Deep Guided Filter into my networks. I want to preserve object boundaries in disparity maps while keeping textures on objects not affecting the disparity estimation. With the default parameters of your `FastGuidedFilter` implementation in PyTorch I get preliminary results like this:
![image](https://user-images.githubusercontent.com/8616975/44629655-0ab24580-a985-11e8-9ec3-c1eeaa4a24bb.png)

where I probably should have set `dgf_eps` to `1e-2` as suggested in your monoDepth code.

The question is as follows:
What do you think if we set this `epsilon` hyperparameter into a learnable one(regulating its range inside (0, 1) with sigmoid etc.) and let the network decide its value? Have you tried this in your experiments?

Thanks for your time.

**Comment by wuhuikai** [comment]
I didn't have an experiment on learnable `eps`. I think that it's enough to determine `eps` at test time. Or you could train a model without `DGF`, determine `eps` with this model and then train the final model with `DGF`.

**Comment by JiamingSuen** [comment]
Thanks for your advice!

**Comment by ynma-hanvo** [comment]
@jiamingSuen   hi, your disparity looks great. may you share which CNN network  you used.  by the way ,  how is the perforamnce in outdoor environment
